.github/scripts/determine-initial-slack-payload.ts

```diff
@@ -0,0 +1,242 @@
+#!/usr/bin/env tsx
+
+import { setOutput } from '@actions/core';
+import fs from 'fs';
+import path from 'path';
+
+type Inputs = {
+  githubRepository: string;
+  githubRunId: string;
+  moduleLintRunsDirectory: string;
+  channelId: string;
+  isRunningOnCi: boolean;
+};
+
+/**
+ * Obtains the inputs for this script from environment variables.
+ *
+ * @returns The inputs for this script.
+ */
+function getInputs(): Inputs {
+  return {
+    githubRepository: requireEnvironmentVariable('GITHUB_REPOSITORY'),
+    githubRunId: requireEnvironmentVariable('GITHUB_RUN_ID'),
+    moduleLintRunsDirectory: requireEnvironmentVariable(
+      'MODULE_LINT_RUNS_DIRECTORY',
+    ),
+    channelId: requireEnvironmentVariable('SLACK_CHANNEL_ID'),
+    isRunningOnCi: process.env.CI !== undefined,
+  };
+}
+
+/**
+ * Obtains the given environment variable, throwing if it has not been set.
+ *
+ * @param variableName - The name of the desired environment variable.
+ * @returns The value of the given environment variable.
+ * @throws if the given environment variable has not been set.
+ */
+function requireEnvironmentVariable(variableName: string): string {
+  const value = process.env[variableName];
+
+  if (value === undefined) {
+    throw new Error(`Missing environment variable ${variableName}.`);
+  }
+
+  return value;
+}
+
+/**
+ * Reads the exit code files produced from previous `module-lint` runs.
+ *
+ * @param moduleLintRunsDirectory - The directory that holds the exit code
+ * files.
+ * @returns An array of exit codes.
+ */
+async function readModuleLintExitCodeFiles(moduleLintRunsDirectory: string) {
+  const entryNames = (await fs.promises.readdir(moduleLintRunsDirectory)).map(
+    (entryName) => path.join(moduleLintRunsDirectory, entryName),
+  );
+  const exitCodeFileNames = entryNames.filter((entry) =>
+    entry.endsWith('--exitcode.txt'),
+  );
+  return Promise.all(
+    exitCodeFileNames.map(async (exitCodeFileName) => {
+      const content = await fs.promises.readFile(exitCodeFileName, 'utf8');
+      return Number(content.trim());
```

Comment:
`Number` sort of does that already, but you're right, this could return `NaN`. I've added a check in [97f42d9](https://github.com/MetaMask/module-lint/pull/91/commits/97f42d9a31ca2624567a5aee174e9727710b0bb1).